xen/arch/arm/domain_build.c

```diff
@@ -1799,6 +1811,15 @@ static int __init handle_node(struct domain *d, struct kernel_info *kinfo,
             return res;
     }
 
+#ifdef CONFIG_SCMI_SMC
+    if ( dt_match_node(scmi_matches, node) )
+    {
+        res = scmi_dt_set_phandle(kinfo, dt_node_full_name(node));
```

Comment:
Why are you passing node name  to that function and then resolving node back? Why you can't pass `node` as is?

Reply:
because it's taken from different device tree